docs/DafnyRef/UserGuide.md

```diff
@@ -101,36 +101,63 @@ on the [Dafny INSTALL page in the wiki](https://github.com/dafny-lang/dafny/wiki
+working directory. A command-line argument not matching a known option is considered a filepathi, and likely one
+with an unsupported suffix, provoking an error message..
 - Files containing dafny code must have a `.dfy` suffix.
 - There must be at least one `.dfy` file.
 - The command-line may contain other kinds of files appropriate to
-the language that the dafny files are being compiled to.
-
-The command `Dafny.exe /?` gives the current set of options supported
-by the tool. The most commonly used options are described in [Section 25.9](#sec-command-line-options).
-
-- Options may begin with either a `/` (as is typical on Windows) or a `-` (as is typical on Linux)
+the language that the Dafny files are being compiled to.
+- Old-style options may begin with either a `/` (as is typical on Windows) or a `-` (as is typical on Linux)
 - If an option is repeated (e.g., with a different argument), then the later instance on the command-line supersedes the earlier instance.
-- If an option takes an argument, the option name is followed by a `:` and then by the argument value, with no
+- If an option takes an argument, the option name is followed by a `:` (old-style) or `=` (new-style) and then by the argument value, with no
 intervening white space; if the argument itself contains white space, the argument must be enclosed in quotes.
 - Escape characters are determined by the shell executing the command-line.
 
+The command `Dafny.exe -?` gives the current set of options supported
+by the tool. The most commonly used options are described in [Section 25.9](#sec-command-line-options).
+
 The `dafny` tool performs several tasks:
 
 - Checking the form of the text in a `.dfy` file. This step is always performed, unless the tool is simply asked for
 help information or version number.
 - Running the verification engine to check all implicit and explicit specifications. This step is performed by
-default, but can be skipped by using the `-noVerify` or `-dafnyVerify:0` option
+default, but can be skipped by using the `-noVerify` or `--no-verify` option
 - Compiling the dafny program to a target language. This step is performed by default if the verification is
-successful but can be skipped or always executed by using variations of the `-compile` option.
-- Whether the source code of the compiled target is written out is controlled by `-spillTargetCode`
-- The particular target language used is chosen by `-compileTarget`
-- Whether or not the `dafny` tool attempts to run the compiled code is controlled by `-compile`
+successful but can be skipped or always executed by using variations of the `-compile` option or different commands.
+- Whether the source code of the compiled target is written out is controlled by `-spillTargetCode` or the top-level command that is being executed.
+- The particular target language used is chosen by `-compileTarget` or `--target`
```

Comment:
This depends on the mode. Do you think it's good to have help both for new and old mode? I imagine we would switch over help to the new mode at some point and then remove help for the old mode.

Reply:
Agreed, but at the moment we need both -- until we deprecate the old style completely.